js/interface.js

```diff
@@ -40,87 +40,67 @@ function selectDataSource(ds) {
   dataSource = ds;
 
   var sms = dataSource.definition && dataSource.definition.validation && dataSource.definition.validation.sms || {};
+  var email = dataSource.definition && dataSource.definition.validation && dataSource.definition.validation.email || {};
   // SMS and email validations use the same expiration values
   // Therefore the value only needs to be restored from the SMS configuration
   $('#expire-timeout').val(sms.expire || defaultExpireTimeout);
   $('#sms-template').val(sms.template || defaultSmsTemplate);
 
+  if (email.domain) {
+    $('#email-domain').prop('checked', true).trigger('change');
+    $('#email-domains').val(email.domains.join(', '));
+  }
+
   Fliplet.Widget.autosize();
 }
 
 var dsQueryData = {};
-if (data.type === 'sms') {
-  dsQueryData = {
-    settings: {
-      dataSourceLabel: 'Select the data source containing the user information',
-      filters: false,
-      columns: [{
-          key: 'smsMatch',
-          label: 'Select the column with the email address',
-          type: 'single'
-        },
-        {
-          key: 'smsTo',
-          label: 'Select the column with the phone number (where the SMS will be sent to)',
-          type: 'single'
-        }
-      ]
-    },
-    result: data.dataSourceQuery
-  };
-} else {
-  dsQueryData = {
-    settings: {
-      dataSourceLabel: 'Select the data source containing the user information',
-      modesDescription: 'Select a verification method',
-      modes: [{
-          label: 'By email',
-          filters: false,
-          columns: [{
-            key: 'emailMatch',
+switch (data.type) {
+  case 'sms':
+    dsQueryData = {
+      settings: {
+        dataSourceLabel: 'Select the data source containing the user information',
+        filters: false,
+        columns: [{
+            key: 'smsMatch',
             label: 'Select the column with the email address',
             type: 'single'
-          }]
-        },
-        {
-          label: 'By domain whitelisting',
-          filters: false,
-          columns: [{
-            key: 'domainMatch',
-            label: 'Select the column with the email address',
+          },
+          {
+            key: 'smsTo',
+            label: 'Select the column with the phone number (where the SMS will be sent to)',
             type: 'single'
-          }]
-        }
-      ]
-    },
-    result: data.dataSourceQuery
-  };
+          }
+        ]
+      },
+      result: data.dataSourceQuery
+    };
+    break;
+
+  case 'email':
+    dsQueryData = {
+      settings: {
+        dataSourceLabel: 'Select the data source containing the user information',
+        filters: false,
+        columns: [{
+          key: 'emailMatch',
+          label: 'Select the column with the email address',
+          type: 'single'
+        }]
+      },
+      result: data.dataSourceQuery
+    };
+    break;
+
+  default:
```

Comment:
What would happen in this case? Will it continue to pass an empty object `{}` to the DS Query provider? Will this create an error? Maybe the error should either be handled, or `default` to the same case as `email`?